common/Environments/Models/ComputeSystem.cs

```diff
@@ -275,6 +275,126 @@ public async Task<ComputeSystemOperationResult> ConnectAsync(string options)
         }
     }
 
+    public async Task<ComputeSystemOperationResult> PinToStartMenuAsync(string options)
+    {
+        try
+        {
+            if (_computeSystem is IComputeSystem2 computeSystem2)
+            {
+                return await computeSystem2.PinToStartMenuAsync();
+            }
+
+            var localErrorString = $"PinToStartMenuAsync for: {this} failed due to exception";
+            var ex = new InvalidOperationException(localErrorString);
+            throw ex;
+        }
+        catch (Exception ex)
+        {
+            _log.Error($"PinToStartMenuAsync for: {this} failed due to exception", ex);
+            return new ComputeSystemOperationResult(ex, errorString, ex.Message);
+        }
+    }
+
+    public async Task<ComputeSystemOperationResult> UnpinFromStartMenuAsync(string options)
+    {
+        try
+        {
+            if (_computeSystem is IComputeSystem2 computeSystem2)
+            {
+                return await computeSystem2.UnpinFromStartMenuAsync();
+            }
+
+            var localErrorString = $"UnpinFromStartMenuAsync for: {this} failed due to exception";

```

Comment:
Can unpining result in an exception?  We use the result method to communicate why something failed when calling a method on the extension.

Reply:
By result method, do you mean returning a *result object like ComputeSystemOperationResult?  If not, I'm not sure what you mean